lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -72,6 +74,19 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+
         for block_key in block_structure.post_order_traversal():
             if block_structure.get_xblock_field(block_key, 'category') not in self.requested_student_view_data:
                 block_structure.remove_transformer_block_field(block_key, self, self.STUDENT_VIEW_DATA)
+            else:
+                student_view_data = block_structure.get_transformer_block_field(
+                    block_key,
+                    self,
+                    self.STUDENT_VIEW_DATA,
+                )
+                block_structure.set_transformer_block_field(
+                    block_key,
+                    self,
+                    self.STUDENT_VIEW_DATA,
+                    student_view_data
+                )
```

Comment:
Aren't the `get_transformer_block_field` and `set_transformer_block_field` calls no-ops - since they should have already been set in the `block_structure` in the `collect` method above?

Reply:
Ach, yes @nasthagiri , you're totally correct.  Sorry for leaving this cruft behind!